src/main/java/com/example/demo/controllers/CustomerController.java

```diff
@@ -0,0 +1,62 @@
+package com.example.demo.controllers;
+
+import com.example.demo.entities.CustomerEntity;
+import com.example.demo.exceptions.CustomerNotFoundException;
+import com.example.demo.exceptions.LocationNotFoundException;
+import com.example.demo.model.dto.CustomerCreationDTO;
+import com.example.demo.services.CustomerService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api")
+public class CustomerController {
+
+    @Autowired
+    CustomerService customerService;
+
+    @PostMapping(path = "/customer")
+    public CustomerEntity createCustomer(@RequestBody CustomerCreationDTO customerDTO)
```

Comment:
To use the `@NotNull` annotation on the name in the entity class you need to add a `@Valid` annotation to the parameter in the controller. We normally add an accompanying test to prove it works too.
```suggestion
    public CustomerEntity createCustomer(@RequestBody @Valid CustomerCreationDTO customerDTO)
```